UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Webjob.UnitTests/Services/FulfilmentAncillaryFilesTest.cs

```diff
@@ -471,38 +471,64 @@ public async Task WhenValidCreateEncUpdateCsvFileRequest_ThenReturnTrueResponse(
         [Test]
         public void WhenInvalidCreateSerialAioFileRequest_ThenReturnFulfilmentException()
         {
-            bool checkAioSerialFileCreated= false;
-            fakeAioExchangeSetPath = @"C:\\HOME";
-            A.CallTo(() => fakeFileSystemHelper.CheckFileExists(A<string>.Ignored)).Returns(false);
+            bool checkAioSerialFileCreated = false;
+
+            A.CallTo(() => fakeFileSystemHelper.CheckFileExists(A<string>.Ignored)).Returns(true).Once().Then.Returns(checkAioSerialFileCreated);
 
             Assert.ThrowsAsync(Is.TypeOf<FulfilmentException>().And.Message.EqualTo(fulfilmentExceptionMessage),
-                  async delegate { await fulfilmentAncillaryFiles.CreateSerialAioFile(fakeBatchId, fakeAioExchangeSetPath, null); });
+                  async delegate { await fulfilmentAncillaryFiles.CreateSerialAioFile(fakeBatchId, fakeExchangeSetRootPath, "", GetSalesCatalogueDataResponse());});
 
             A.CallTo(fakeLogger).Where(call => call.Method.Name == "Log"
             && call.GetArgument<LogLevel>(0) == LogLevel.Error
             && call.GetArgument<EventId>(1) == EventIds.SerialAioFileIsNotCreated.ToEventId()
             && call.GetArgument<IEnumerable<KeyValuePair<string, object>>>(2)!.ToDictionary(c => c.Key, c => c.Value)["{OriginalFormat}"].ToString() == "Error in creating serial.aio file for BatchId:{BatchId} and _X-Correlation-ID:{CorrelationId} - Invalid Exchange Set Path").MustHaveHappenedOnceExactly();
 
-            Assert.AreEqual(false, fakeFileHelper.CheckAndCreateFolderIsCalled);
             Assert.AreEqual(false, checkAioSerialFileCreated);
+
+            A.CallTo(() => fakeFileSystemHelper.CheckAndCreateFolder(A<string>.Ignored)).MustHaveHappenedOnceExactly();
+            A.CallTo(() => fakeFileSystemHelper.CreateFileContent(A<string>.Ignored, A<string>.Ignored)).MustHaveHappenedOnceExactly();
         }
 
         [Test]
         public async Task WhenValidCreateSerialAioFileRequest_ThenReturnTrueResponse()
         {
             bool checkAioSerialFileCreated = true;
-            fakeAioExchangeSetPath = @"C:\\HOME";
-            fakeFileHelper.CheckAndCreateFolder(fakeExchangeSetInfoPath);
+                        
+            A.CallTo(() => fakeFileSystemHelper.CheckFileExists(A<string>.Ignored)).Returns(checkAioSerialFileCreated).Twice();
 
-            A.CallTo(() => fakeFileSystemHelper.CheckAndCreateFolder(A<string>.Ignored));
-            A.CallTo(() => fakeFileSystemHelper.CreateFile(A<string>.Ignored));
-            A.CallTo(() => fakeFileSystemHelper.CheckFileExists(A<string>.Ignored)).Returns(true);
+            checkAioSerialFileCreated = await fulfilmentAncillaryFiles.CreateSerialAioFile(fakeBatchId, fakeExchangeSetRootPath, "", GetSalesCatalogueDataResponse());
 
-            var response = await fulfilmentAncillaryFiles.CreateSerialAioFile(fakeBatchId, fakeAioExchangeSetPath, null);
+            Assert.AreEqual(true, checkAioSerialFileCreated);
```

Comment:
```suggestion
            Assert.IsTrue(checkAioSerialFileCreated);
```
Please use suggested way while checking boolean. Similarly follow for false scenario.

Reply:
Resolved